pkg/puller/puller.go

```diff
@@ -53,13 +66,15 @@ type Puller struct {
 }
 
 func New(o Options) *Puller {
+	bins = o.Bins
+	shallowBinPeers = o.ShallowBinPeers
```

Comment:
This constructor is setting global variables. How would that affect other Puller instances? There will be a data race when reading these global variables in manage loop and creating a new Puller instance.

Reply:
why don't we put these global variables as `Puller` fields?